nntrainer/layers/common_properties.h

```diff
@@ -316,6 +551,191 @@ class NumClass final : public nntrainer::Property<unsigned int> {
    */
   bool isValid(const unsigned int &v) const override;
 };
+
+/******** below section is for enumerations ***************/
+/**
+ * @brief     Enumeration of activation function type
+ */
+struct ActivationTypeInfo {
+  using Enum = nntrainer::ActivationType;
+  static constexpr std::initializer_list<Enum> EnumList = {
+    Enum::ACT_TANH,    Enum::ACT_SIGMOID, Enum::ACT_RELU,
+    Enum::ACT_SOFTMAX, Enum::ACT_NONE,    Enum::ACT_UNKNOWN};
+
+  static constexpr const char *EnumStr[] = {"tanh",    "sigmoid", "relu",
+                                            "softmax", "none",    "unknown"};
+};
+
+/**
+ * @brief Activation Enumeration Information
+ *
+ */
+class Activation final : public EnumProperty<ActivationTypeInfo> {
+public:
+  using prop_tag = enum_class_prop_tag;
+  static constexpr const char *key = "activation";
+};
+
+/**
+ * @brief HiddenStateActivation Enumeration Information
+ *
+ */
+class HiddenStateActivation final : public EnumProperty<ActivationTypeInfo> {
+public:
+  /**
+   * @brief Construct a new HiddenStateActivation object with default value
+   * ActivationTypeInfo::Enum::ACT_NONE
+   *
+   */
+  HiddenStateActivation(
+    ActivationTypeInfo::Enum value = ActivationTypeInfo::Enum::ACT_NONE);
+  using prop_tag = enum_class_prop_tag;
+  static constexpr const char *key = "hidden_state_activation";
+};
+
+/**
+ * @brief RecurrentActivation Enumeration Information
+ *
+ */
+class RecurrentActivation final : public EnumProperty<ActivationTypeInfo> {
+public:
+  /**
+   * @brief Construct a new RecurrentActivation object with default value
+   * ActivationTypeInfo::Enum::ACT_NONE
+   *
+   */
+  RecurrentActivation(
+    ActivationTypeInfo::Enum value = ActivationTypeInfo::Enum::ACT_NONE);
+  using prop_tag = enum_class_prop_tag;
+  static constexpr const char *key = "recurrent_activation";
+};
+
+/**
+ * @brief     Enumeration of tensor initialization type
+ */
+struct InitializerInfo {
+  using Enum = Tensor::Initializer;
+  static constexpr std::initializer_list<Enum> EnumList = {
+    Enum::ZEROS,         Enum::ONES,          Enum::LECUN_NORMAL,
+    Enum::LECUN_UNIFORM, Enum::XAVIER_NORMAL, Enum::XAVIER_UNIFORM,
+    Enum::HE_NORMAL,     Enum::HE_UNIFORM,    Enum::NONE};
+
+  static constexpr const char *EnumStr[] = {
+    "zeros",         "ones",          "lecun_normal",
+    "lecun_uniform", "xavier_normal", "xavier_uniform",
+    "he_normal",     "he_uniform",    "none"};
+};
+
+/**
+ * @brief BNPARAMS_MU Initialization Enumeration Information
+ *
+ */
+class BNPARAMS_MU final : public EnumProperty<InitializerInfo> {
```

Comment:
recommend to update property name to BNPARAMS_MU_INIT and for other BNPARAMS as well.

Reply:
Done